analyzer/src/funTest/kotlin/integration/AbstractIntegrationSpec.kt

```diff
@@ -0,0 +1,118 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package com.here.ort.analyzer.integration
+
+import com.here.ort.analyzer.ManagedProjectFiles
+import com.here.ort.analyzer.PackageManager
+import com.here.ort.downloader.Main
+import com.here.ort.downloader.VersionControlSystem
+import com.here.ort.model.Package
+import com.here.ort.utils.Expensive
+
+import io.kotlintest.Spec
+import io.kotlintest.matchers.beEmpty
+import io.kotlintest.matchers.shouldBe
+import io.kotlintest.matchers.shouldNot
+import io.kotlintest.matchers.shouldNotBe
+import io.kotlintest.specs.StringSpec
+
+import java.io.File
+
+abstract class AbstractIntegrationSpec : StringSpec() {
+
+    /**
+     * The software package to download.
+     */
+    protected abstract val pkg: Package
+
+    /**
+     * The definition files that are expected to be found by the [PackageManager].
+     */
+    protected abstract val expectedDefinitionFiles: ManagedProjectFiles
+
+    /**
+     * The definition files that shall be used for dependency resolution. Defaults to [expectedDefinitionFiles], but
+     * can be reduced for large projects that have a lot of definition files to speed up the test.
+     */
+    protected open val definitionFilesForTest by lazy { expectedDefinitionFiles }
+
+    /**
+     * The directory where the source code of [pkg] was downloaded to.
+     */
+    protected lateinit var downloadDir: File
+
+    /**
+     * A temporary directory used as working directory for the test suite.
+     */
+    private val outputDir = createTempDir()
+
+    /**
+     * The instance needs to be shared between tests because otherwise the source code of [pkg] would have to be
+     * downloaded once per test.
+     */
+    override val oneInstancePerTest = false
+
+    override fun interceptSpec(context: Spec, spec: () -> Unit) {
+        spec()
+        outputDir.deleteRecursively()
+    }
+
+    init {
+        "Source code can be downloaded" {
+            downloadDir = Main.download(pkg, outputDir)
+            val workingDir = VersionControlSystem.forDirectory(downloadDir)
+
+            downloadDir shouldNotBe null
+            workingDir shouldNotBe null
+            workingDir!!.isValid() shouldBe true
+            workingDir.getProvider() shouldBe pkg.vcs.provider
+        }.config(tags = setOf(Expensive))
+
+        "All package manager definition files are found" {
+            val definitionFiles = PackageManager.findManagedFiles(downloadDir)
+
+            definitionFiles.size shouldBe expectedDefinitionFiles.size
```

Comment:
Shouldn't `expectedDefinitionFiles.size` be `definitionFilesForTest.size`?

Reply:
No, expectedDefinitionFiles is for this test, while definitionFilesForTest is for the dependency resolution test below.